### ISSUE TITLE
Exception during build: "Could not get unknown property 'sourceSets'"

### ISSUE THREAD

**Issue by apryamostanov** [virtualdogbert/enterprise-groovy-plugin#1]
Hi @virtualdogbert ,

I am getting attached exception when running attached build.gradle.
`Exception during build: "Could not get unknown property 'sourceSets'"`
[exception.txt](https://github.com/virtualdogbert/enterprise-groovy-plugin/files/2840210/exception.txt)
[build.gradle](https://github.com/virtualdogbert/enterprise-groovy-plugin/files/2840213/build.txt)

Repository: https://github.com/INFINITE-TECHNOLOGY/SUPPLIES

Would you be so kind to check this error please?

Thank you.

**Comment by virtualdogbert** [comment]
I'll take a look after work tonight. Thanks.

**Comment by apryamostanov** [comment]
Just one small note: I didn't run the setup task. Is it mandatory to have conventions.groovy?

**Comment by virtualdogbert** [comment]
I think I do have defaults for everything, but it would be better to at least run the default setup task, so that you can configure with the conventions.groovy.

For the other issue I'm wondering if it might be related to a gradle version thing... One thing I'm thinking of trying if adding:
apply plugin: 'groovy'

instead of the gradle 5 way, in your project. In either case, I'm going to add, a optional gradle property so that you can disable the Groovy Console task is it causes issues.

**Comment by virtualdogbert** [comment]
I've been working on it and part of it was an issue, with gradle, and once I got past that I found of other issues,  which link back to the gradle version. I'm going to play with it some more, see if I can resolve the last gradle issue and put out an RC2 this weekend.

The last gradle issue I have is with the config file. If I'm using the gradle wrapper version 5.0, it doesn't see the config file,but if I use 4.10.2, the same version I compiled the plugin under, then I can see the config file. I've also refactored the code so that if there is no config file then you get the equivalent of the default config.

**Comment by virtualdogbert** [comment]
I ran into several problems with Gradle 5. I fixed it so that the config and the console will work, however the plugin itself will only work if applied the old way using the apply plugin, rather than the newer plugin dsl. I'm still going to look into getting that fixed, but for right now I can't find a way to add a library to the project, that Enterprise Groovy is applied to that works. More research is required...
